src/components/components/AddLayerPanel/AddLayerPanel.component.jsx

```diff
@@ -0,0 +1,378 @@
+      if (inSegment) {
+        preEntity.setAttribute('position', selectedObjPos);
+      } else {
+        preEntity.setAttribute('position', {x: selectedObjPos.x, y: 0.2, z: selectedObjPos.z});
+      }
+    }
+  };
+
+  const cardMouseLeave = (mixinId) => {
+    preEntity.setAttribute('visible', false);
+  };
+
+  const createEntity = (mixinId, parentEl) => {
+    console.log('create entity: ', mixinId);
+    const newEntity = document.createElement('a-entity');
+    newEntity.setAttribute('mixin', mixinId);
+
+    const selectedElement = AFRAME.INSPECTOR.selected?.el;
+
+    // avoid adding new element inside the direct ancestor of a-scene: #environment, #reference, ...
+    if (selectedElement && !ancestorOfSelectedEl.parentEl.isScene) {
+      // append element as a child of the entity with .custom-group class.
+      let customGroupEl = ancestorOfSelectedEl.querySelector('.custom-group');
+      if (!customGroupEl) {
+        customGroupEl = document.createElement('a-entity');
+        // .custom-group entity is a child of segment or .street-parent/.buildings-parent elements
+        ancestorOfSelectedEl.appendChild(customGroupEl);
+        customGroupEl.classList.add('custom-group');
+
+        if (inSegment) {
+          // set position y by elevation level of segment
+          customGroupEl.setAttribute('position', { y: segmentElevationPosY });
+        } else {
+          // if we are creating element not inside segment-parent
+          customGroupEl.setAttribute('position', selectedObjPos);
+        }
+      }
+      customGroupEl.appendChild(newEntity);
+    } else {
+      const streetContainer = document.querySelector('street-container');
+      // apppend element as a child of street-container
+      if (streetContainer) {
+        streetContainer.appendChild(newEntity);
+      } else {
+        AFRAME.scenes[0].appendChild(newEntity);
+      }
+    }
+    Events.emit('entitycreated', newEntity);
+  };
+
+  const cardClick = (card) => {
+    if (card.mixinId) {
+      createEntity(card.mixinId);
+    } else if (card.handlerFunction) {
+      card.handlerFunction();
+    }
+  };
+  return (
+    <div
+      className={classNames(styles.panel, isAddLayerPanelOpen && styles.open)}
+    >
+      <Button onClick={onClose} variant="custom" className={styles.closeButton}>
+        <Chevron24Down />
+      </Button>
+      <div className={styles.header}>
+        <div className={styles.button}>
+          <Plus20Circle />
+          <p className={styles.buttonLabel}>Add New Entity</p>
+        </div>
+        <Dropdown
+          placeholder="Layers: Maps & Reference"
+          options={options}
+          onSelect={handleSelect}
+          selectedOptionValue={selectedOption}
+          className={styles.dropdown}
+          smallDropdown={true}
+        />
+      </div>
+      <div className={styles.cards}>
+        {selectedCards?.map((card, idx) => (
+          <div
+            key={Number(card.id + idx)}
```

Comment:
You don't need to do this, just use `key={idx}`, key just need to be unique inside the map, not globally.

Reply:
Or `key={card.id}`, better to use an identifier than an index when you can. `Number(card.id + idx)` is probably returning `NaN`, for example `Number("card1"+ 2)` return `NaN`.